services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/MetersSpec.groovy

```diff
@@ -0,0 +1,219 @@
+
+        then: "New meters should be installed on the switch"
+        def newMeters = northboundService.getAllMeters(sw.dpId).meterEntries.findAll {
+            !defaultMeters.meterEntries.contains(it)
+        }
+        newMeters.size() == 2
+
+        and: "New meters rate should be equal to flow bandwidth"
+        newMeters*.rate.every { it == flowRate }
+
+        and: "New meters burst size should be between 100.5% and 105% of the flow's rate"
+        newMeters*.burstSize.every { (it >= flowRate * NOVI_BURST_MIN) && (it <= flowRate * NOVI_BURST_MAX) }
+
+        cleanup: "Delete the flow"
+        northboundService.deleteFlow(flow.id)
+
+        where:
+        flowRate << [150, 1000, 1024, 5120, 10240, 2480, 960000]
+    }
+
+    def "Flow burst should not exceed 1024 kbps on Centec switches"() {
+        setup: "A flow with #flowRate kbps bandwidth is created on OpenFlow 1.3 compatible switch"
+        this.requireProfiles("hardware")
+        def sw = getNonCentecSwitches()[0]
+        def defaultMeters = northboundService.getAllMeters(sw.dpId)
+        def flowPayload = flowHelper.singleSwitchFlow(sw)
+        flowPayload.setMaximumBandwidth(10)
+        def flow = flowHelper.addFlow(flowPayload)
+
+        when: "Update flow bandwidth to #flowRate kbps"
+        flow = northboundService.getFlow(flow.id)
+        flow.setMaximumBandwidth(flowRate)
+        northboundService.updateFlow(flow.id, flow)
+        sleep(Constants.RULES_INSTALLATION_TIME * 1000) //TODO: add helper method to actually monitor this
+
+        then: "New meters should be installed on the switch"
+        def newMeters = northboundService.getAllMeters(sw.dpId).meterEntries.findAll {
+            !defaultMeters.meterEntries.contains(it)
+        }
+        newMeters.size() == 2
+
+        and: "New meters rate should be equal to flow bandwidth"
+        newMeters*.rate.every { it == flowRate }
+
+        and: "New meters burst size should be 1024 kbit/s regardless the flow speed"
+        newMeters*.burstSize.every { it == burstSize }
+
+        cleanup: "Delete the flow"
+        northboundService.deleteFlow(flow.id)
+
+        where:
+        flowRate | burstSize
+        100      | 1024
+        1024     | 1024
+        20480    | 1024
+        960000   | 1024
+    }
+
+    def "Kilda should be able to delete a meter"() {
+        setup: "Select a switch and retrieve default meters"
+        def sw = topology.getActiveSwitches()[0]
+        def defaultMeters = northboundService.getAllMeters(sw.dpId)
+
+        when: "A flow is created and its meter is deleted"
+        def flow = flowHelper.addFlow(flowHelper.singleSwitchFlow(sw))
+        def meterToDelete = northboundService.getAllMeters(sw.dpId).meterEntries.find {
+            !defaultMeters.meterEntries*.meterId.contains(it.meterId)
+        }.meterId
+        def deleteResult = northboundService.deleteMeter(sw.dpId, meterToDelete)
+
+        then: "Delete operation should be successful"
```

Comment:
since this test is fully focused on 'deletion', I think it makes sense to verify that the meter is absent in the results of the subsequent 'get' request.

Reply:
Added the verification